public/js/actions/notifications.js

```diff
@@ -1,14 +1,58 @@
 import * as actionTypes from 'constants/action-types';
+import * as errorCodes from 'constants/error-codes';
 
+import { gettext } from 'utils';
 
-export function addNotification(text, { type='info', userDismissable=false,
-                                autoHide=true } = {}) {
+
+export function addNotification({ text, errorCode,
+                                  type='info', userDismissable,
+                                  autoHide } = {}) {
+
+  if (type === 'error' && !text) {
+    text = gettext('Internal error. Please try again later.');
+  }
+
+  if (type === 'info' && !text) {
```

Comment:
I think this could just be `if (!text)...` because `text` is set up above for error types